File: hw2/Tests2/ErrorMessagesTests.cs
using hw2_il;
using Xunit;

namespace Tests
{
    public class ErrorMessagesTests
    {
        [Fact]
        public void IsErrorCodeDisplayErrorMessage_WrongInput_WillReturnFalse()
        {
            var args = new[] { "1", "+", "2" };
            var result = ErrorMessages.IsErrorCodeDisplayErrorMessage(ErrorCode.Correct, args);
            Assert.False(result);
        }
        
        [Theory]
        [InlineData(1, "1", "+", "1", "")]
        [InlineData(2, "1", "+", "s")]
        [InlineData(3, "1", "fd", "1")]
        public void IsErrorCodeDisplayErrorMessage_WrongInput_WillReturnTrue(int code, params string[] args)
        {
            var result = ErrorMessages.IsErrorCodeDisplayErrorMessage((ErrorCode) code, args);
            Assert.True(result);
        }
    }
}

File: hw3/Tests3/ErrorMessagesTests.fs
module Tests.ErrorMessagesTests
open Xunit
open hw3

[<Fact>]
let ``IsErrorCodeDisplayErrorMessage_WrongInput_WillReturnFalse`` () =
    let args = [|"1";"+";"2"|]
    let result = ErrorMessages.isErrorCodeDisplayErrorMessage ErrorCode.Correct args
    Assert.False(result)
        
[<Theory>]
[<InlineData(ErrorCode.WrongArgLength, "1", "+", "1", true)>]
[<InlineData(ErrorCode.WrongArgFormatInt, "1", "+", "s", false)>]
[<InlineData(ErrorCode.WrongArgFormatOperation, "1", "fd", "1", false)>]
let ``IsErrorCodeDisplayErrorMessage_WrongInput_WillReturnTrue`` (code, arg1, arg2, arg3, isFourArgs) =
    if isFourArgs then
        let result = ErrorMessages.isErrorCodeDisplayErrorMessage code [|arg1;arg2;arg3;""|]
        Assert.True(result)
    else
        let result = ErrorMessages.isErrorCodeDisplayErrorMessage code [|arg1;arg2;arg3|]
        Assert.True(result);